brainscore_language/models/earley_parser/test.py

```diff
@@ -0,0 +1,242 @@
+                Det  -> 'a'           [.31]
+                Det  -> 'my'          [.28]
+            """
+
+
+# Grammars estimated from sample_treebank with different hyperparameters
+sample_grammars_hparam_keys = ("k",)
+sample_grammars_by_hparams = {
+    (2,): """
+        S -> NP VP [1.0]
+        NP -> Det Adj N [0.45]
+        Det -> 'a' [0.55]
+        Adj -> 'brown' [0.444444]
+        N -> 'fox' [0.15]
+        VP -> V NP [0.363636]
+        V -> 'saw' [0.363636]
+        NP -> Det N [0.55]
+        N -> 'man' [0.45]
+        Det -> 'the' [0.45]
+        VP -> V PP [0.454545]
+        V -> 'walked' [0.181818]
+        PP -> P NP [1.0]
+        P -> 'with' [0.4]
+        N -> 'dog' [0.4]
+        V -> 'jumped' [0.272727]
+        P -> 'over' [0.4]
+        Adj -> 'lazy' [0.555556]
+        VP -> V [0.181818]
+        V -> 'slept' [0.181818]
+        P -> '<unk>' [0.2]
+        """,
+
+    (3,): """
+        S -> NP VP [1.0]
+        NP -> Det Adj N [0.45]
+        Det -> 'a' [0.55]
+        Adj -> 'brown' [0.444444]
+        N -> 'fox' [0.15]
+        VP -> V NP [0.363636]
+        V -> 'saw' [0.363636]
+        NP -> Det N [0.55]
+        N -> 'man' [0.45]
+        Det -> 'the' [0.45]
+        VP -> V PP [0.454545]
+        V -> '<unk>' [0.363636]
+        PP -> P NP [1.0]
+        P -> '<unk>' [1.0]
+        N -> 'dog' [0.4]
+        V -> 'jumped' [0.272727]
+        Adj -> 'lazy' [0.555556]
+        VP -> V [0.181818]
+        """,
+}
+
+
+@pytest.mark.parametrize(
+    "model_identifier, treebank_path, hparams, expected_grammar",
+    [
+        (
+            "earley-parser",
+            str(Path(__file__).parent / "treebank"),
+            dict(zip(sample_grammars_hparam_keys, hparams)),
+            grammar
+        )
+        for hparams, grammar in sample_grammars_by_hparams.items()
+    ]
+)
+def test_create_grammar(model_identifier, treebank_path, hparams, expected_grammar):
+    """
+    Estimate a PCFG grammar from a treebank with different hyperparameters.
+    """
+    model: EarleyParserSubject = load_model(model_identifier)
+    model.create_grammar(
+        treebank_path=treebank_path,
+        fileids="sample_treebank",
+        **hparams,
+    )
+
+    expected_grammar = PCFG.fromstring(expected_grammar)
+
+    # HACK: Round off estimated probabilities to 6 decimal places to match
+    # ground-truth grammars.
+    estimated_productions = [
+        ProbabilisticProduction(p.lhs(), p.rhs(), prob=round(p.prob(), 6))
+        for p in model.grammar.productions()
+    ]
+    assert set(estimated_productions) == set(expected_grammar.productions())
+
+
+@pytest.mark.parametrize(
+    "model_identifier, treebank_path, expected_reading_times, expected_next_words",
+    [
+        (
+            "earley-parser",
+            str(Path(__file__).parent / "treebank"),
+            [1.15200, 3.4739, 6.2109, 8.08537, 9.4073, 10.5593, 12.5593, 13.8812],
+            ["lazy", "man", "saw", "with", "a", "lazy", "man", "<unk>"],
```

Comment:
I would prefer to separate this into two tests, one for reading times and one for next words. This might be personal preference (I like detailed test reports), so if you think this is the way to go, feel free to ignore.